hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -1641,7 +1641,7 @@ void optimizeARM(Vunit& unit, const Abi& abi, bool regalloc) {
   simplify(unit);
 
   lowerForARM(unit);
-
+  reuseImmq(unit);
```

Comment:
I think it would be better to call this after foldImms(). foldImms() eliminates immediates a bit more efficient than this pass.